### DISCUSSION TITLE
Import error ( probably circular import)

### DISCUSSION THREAD

**Discussion by Kouroshdan** [python-telegram-bot/python-telegram-bot#4361] [Q&A]
hey , when I try to run my code for a bot a got this error 

```python
line 1, in <module>
    from telegram import Update, InlineKeyboardButton, InlineKeyboardMarkup
  File "C:\Users\Classic\AppData\Local\Programs\Python\Python311\Lib\site-packages\telegram\__init__.py", line 249, in <module>
    from ._bot import Bot
  File "C:\Users\Classic\AppData\Local\Programs\Python\Python311\Lib\site-packages\telegram\_bot.py", line 85, in <module>     
    from telegram._message import Message
  File "C:\Users\Classic\AppData\Local\Programs\Python\Python311\Lib\site-packages\telegram\_message.py", line 2, in <module>  
    from telegram import Update, InlineKeyboardButton, InlineKeyboardMarkup
ImportError: cannot import name 'Update' from partially initialized module 'telegram' (most likely due to a circular import) (C:\Users\Classic\AppData\Local\Programs\Python\Python311\Lib\site-packages\telegram\__init__.py)
```

how can I solve this ? If needed I can share my code , tnx   (note , my python-telegram-bot version  is 21.3)

**Comment by Bibo-Joshi** [comment]
Please have a look at https://github.com/python-telegram-bot/python-telegram-bot/wiki/Frequently-Asked-Questions#why-am-i-getting-importerror-cannot-import-name-xy-from-telegram. If the tips there don't help you, please report it here.

**Comment by harshil21** [comment]
> ```
>   File "C:\Users\Classic\AppData\Local\Programs\Python\Python311\Lib\site-packages\telegram\_message.py", line 2, in <module>  
>     from telegram import Update, InlineKeyboardButton, InlineKeyboardMarkup
> ImportError: cannot import name 'Update' from partially initialized module 'telegram' (most likely due to a circular import) (C:\Users\Classic\AppData\Local\Programs\Python\Python311\Lib\site-packages\telegram\__init__.py)
> ```

Sorry but that line just does not exist in our file: https://github.com/python-telegram-bot/python-telegram-bot/blob/v21.3/telegram/_message.py, so you've either edited that file or I don't know what.
